File: backend/internal/models/setting.go
package models

import "time"

type Setting struct {
	ID          uint      `json:"id" gorm:"primaryKey"`
	Key         string    `json:"key" gorm:"unique;not null"`
	Value       string    `json:"value" gorm:"type:text"`
	Type        string    `json:"type" gorm:"default:'text'"` // text, textarea, boolean, number, image
	Description string    `json:"description"`
	Group       string    `json:"group" gorm:"default:'general'"` // general, seo, contact, social
	IsPublic    bool      `json:"is_public" gorm:"default:false"` // If true, returned in public API
	CreatedAt   time.Time `json:"created_at"`
	UpdatedAt   time.Time `json:"updated_at"`
}

File: frontend/src/types/setting.ts
export interface Setting {
    id: number;
    key: string;
    value: string;
    type: 'text' | 'textarea' | 'boolean' | 'number' | 'image';
    description: string;
    group: string;
    is_public: boolean;
    created_at: string;
    updated_at: string;
}

export interface UpdateSettingInput {
    key: string;
    value: string;
    is_public?: boolean;
}
